test/unit/submission/validation.spec.ts

```diff
@@ -1444,6 +1445,98 @@ describe('data-validator', () => {
 
       chai.expect(result.specimen.dataErrors).to.deep.include(specimenMissingDonorAB2FieldErr);
     });
+    it('should detect forbidden provided fields', async () => {
+      const existingDonorAB1Mock: Donor = stubs.validation.existingDonor04();
+
+      const forbiddenFields = [
+        SpecimenFieldsEnum.tumour_grading_system,
+        SpecimenFieldsEnum.tumour_grade,
+        SpecimenFieldsEnum.pathological_tumour_staging_system,
+        SpecimenFieldsEnum.pathological_stage_group,
+        SpecimenFieldsEnum.percent_stromal_cells,
+        SpecimenFieldsEnum.percent_tumour_cells,
+        SpecimenFieldsEnum.percent_proliferating_cells,
+        SpecimenFieldsEnum.percent_inflammatory_tissue,
+        SpecimenFieldsEnum.percent_necrosis,
+        SpecimenFieldsEnum.pathological_m_category,
+        SpecimenFieldsEnum.pathological_t_category,
+        SpecimenFieldsEnum.pathological_n_category,
+        SpecimenFieldsEnum.tumour_histological_type,
+        SpecimenFieldsEnum.central_pathology_confirmed,
+      ];
+
+      const newDonorAB1Records = {};
+      const specimenFields = _.fromPairs(
+        forbiddenFields.map(field => [field, 'a forbidden provided value']),
+      );
+
+      ClinicalSubmissionRecordsOperations.addRecord(
+        ClinicalEntitySchemaNames.SPECIMEN,
+        newDonorAB1Records,
+        {
+          [SampleRegistrationFieldsEnum.submitter_donor_id]: 'AB2',
+          [SampleRegistrationFieldsEnum.submitter_specimen_id]: 'SP13',
+          [SpecimenFieldsEnum.specimen_acquisition_interval]: 400,
+          index: 0,
+          ...specimenFields,
+        },
+      );
+      const result = await dv
+        .validateSubmissionData({ AB1: newDonorAB1Records }, { AB1: existingDonorAB1Mock })
+        .catch((err: any) => fail(err));
+
+      chai.expect(result.specimen.dataErrors.length).to.eq(forbiddenFields.length);
+      result.specimen.dataErrors.forEach(dataError => {
+        chai.expect(forbiddenFields).includes(dataError.fieldName);
+        chai
+          .expect(dataError.type)
+          .to.equal(DataValidationErrors.FORBIDDEN_PROVIDED_VARIABLE_REQUIREMENT);
+      });
+    });
+    it('should detect missing required fields', async () => {
```

Comment:
this description makes it sound like its a general schema validation rather then a specific data validation. Maybe `should detect missing required fields in submitted record for registered donor tumour specimen`